doc/tools/make_rst.py

```diff
@@ -995,17 +988,21 @@ def rstize_text(text, state):  # type: (str, State) -> str
             elif cmd.find("image=") == 0:
                 tag_text = ""  # '![](' + cmd[6:] + ')'
             elif cmd.find("url=") == 0:
-                url_link = cmd[4:]
-                tag_text = "`"
-                tag_depth += 1
-                inside_url = True
-                url_has_name = False
-            elif cmd == "/url":
-                tag_text = ("" if url_has_name else url_link) + " <" + url_link + ">`__"
-                tag_depth -= 1
-                escape_post = True
-                inside_url = False
-                url_has_name = False
+                # URLs are handled in full here as we need to extract the optional link
+                # title to use `make_link`.
+                link_url = cmd[4:]
+                endurl_pos = text.find("[/url]", endq_pos + 1)
+                if endurl_pos == -1:
+                    print_error("Tag depth mismatch for [url]: no closing [/url], file: {}".format(state.current_class), state)
+                    break
+                link_title = text[endq_pos + 1 : endurl_pos]
+                tag_text = make_link(link_url, link_title)
+
+                pre_text = text[:pos]
+                text = pre_text + tag_text + text[endurl_pos + 6 :]
+                pos = len(pre_text) + len(tag_text)
+                previous_pos = pos
+                continue
```

Comment:
This is a bit ugly and not the correct way to do this in a parser I suppose... but I couldn't figure out a cleaner way without bypassing the loop. CC @pycbouh